src/bot/index.ts

```diff
@@ -106,36 +101,19 @@ bot.on("Room.timeline", (event: mSDK.MatrixEvent) => {
   const [action, arg0, arg1] = body.split(" ");
 
   if (action === "!version") {
-    ax.get<APIVersionResponse>("/version")
-      .then((res) => {
-        const { data } = res;
-        const { version, time } = data;
-
-        sendMessage(
-          roomId,
-          `Versions:
-          Bot: ${botDeployedRef}; ${botDeployedTime}
-          Server: ${version}; ${time}`,
-        );
-      })
-      .catch((e) => {
-        sendMessage(roomId, "An error occurred, please check the server logs.");
-        logger.error("⭕ An error occurred when checking the balance", e);
-      });
+    sendMessage(roomId, `Current version: ${deployedRef}; ${deployedTime}`);
   } else if (action === "!balance") {
-    ax.get<BalanceResponse>("/balance")
-      .then((res) => {
-        const balance = Number(res.data.balance);
-
-        sendMessage(
-          roomId,
-          `The faucet has ${balance / 10 ** networkData.decimals} ${networkData.currency}s remaining.`,
-        );
-      })
-      .catch((e) => {
-        sendMessage(roomId, "An error occurred, please check the server logs.");
-        logger.error("⭕ An error occurred when checking the balance", e);
-      });
+    (async () => {
+      const balance = await polkadotActions.getBalance();
+
+      sendMessage(
+        roomId,
+        `The faucet has ${Number(balance) / 10 ** networkData.decimals} ${networkData.currency}s remaining.`,
+      );
+    })().catch((e) => {
+      sendMessage(roomId, "An error occurred, please check the server logs.");
+      logger.error("⭕ An error occurred when checking the balance", e);
+    });
```

Comment:
Could this maybe be replaced for a try/catch block?

As it is no longer calling a function, it could be cleaner to read maybe?

Reply:
The surrounding function (the callback for `MatrixClient#on`) is synchronous